main/guides/js-programming/hardened-js.md

```diff
@@ -0,0 +1,388 @@
+
+This limits the damage that can happen if there is an exploitable bug.
+
+::: tip Watch: Navigating the Attack Surface
+to achieve a *multiplicative* reduction in risk<br />
+[![Navigating the Attack Surface](https://i.ytimg.com/vi/wW9-KuezPp8/hqdefault.jpg?sqp=-oaymwEcCPYBEIoBSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLBBrRPtKSnPu3Quq6yyJeJdu1kEyQ)](https://www.youtube.com/watch?v=wW9-KuezPp8)<br />
+_15 min_
+:::
+
+## Tool Support: eslint config
+
+::: tip eslint configuration for Jessie
+The examples that follow are written using _Jessie_, our
+recommended style for writing JavaScript smart contracts.
+This `eslint` configuration provides tool support.
+:::
+
+1. If not already configured, run `yarn add eslint @jessie.js/eslint-plugin`
+2. If not already configured, add the following to your `package.json`:
+
+```json
+  "eslintConfig": {
+    "extends": [
+      "@jessie.js"
+    ]
+  }
+```
+
+3. Put `// @jessie-check` at the beginning of your `.js` source file.
+4. Run `yarn eslint --fix path/to/your-source.js`
+5. Follow the linter's advice to edit your file, then go back to step 4.
+
+The example in the video cuts a few corners to better
+fit on a slide. For clarity and consistency
+with [Airbnb JavaScript Style](https://github.com/airbnb/javascript#variables--unary-increment-decrement), we recommend:
+ - Avoid `--` and `++` operators.
+ - Break lines after `{` and before `}`.
+
+## Objects and the _maker_ pattern
+
+Let's unpack the `makeCounter` example a bit.
+
+JavaScript is somewhat novel in that objects need not belong to any
+class; they can just stand on their own:
+
+<<< @/snippets/test-hardened-js.js#singleton
+
+We can make a new such object each time a function is called
+using the _maker pattern_:
+
+<<< @/snippets/test-hardened-js.js#maker
+
+::: tip Use lexically scoped variables rather than properties of this.
+The style above avoids boilerplate such as `this.x = x; this.y = y`.
+:::
+
+::: tip Use arrow functions
+We recommend [arrow function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions)
+syntax rather than `function makePoint(x, y) { ... }` declarations
+for conciseness and to avoid `this`.
+:::
+
+## Defensive objects with `harden()`
+
+By default, anyone can clobber the properties of
+our objects so that they fail to conform to the expected API:
+
+<<< @/snippets/test-hardened-js.js#clobber
+
+Worse yet is to clobber a property so that it misbehaves but
+covers its tracks so that we don't notice:
+
+<<< @/snippets/test-hardened-js.js#exploit
+
+Our goal is **defensive correctness**: a program is _defensively correct_ if it remains correct despite arbitrary behavior on the part of its clients. _For further discussion, see [Concurrency Among Strangers](http://erights.org/talks/promises/paper/tgc05.pdf) and other [Agoric papers on Robust Composition](https://papers.agoric.com/papers/#robust-composition)_.
```

Comment:
I don't understand why, but the tg05 url works manually, but not when I click on it in the rendered text.

Reply:
Hm. I can't seem to reproduce the problem.